### PR TITLE
Filter shopping list with input field

### LINKED ISSUE #9: 9. As a user, I want to filter my shopping list to make it easier to locate an item in the list.
AC:
      * Display a text field above the top of the shopping list
      * As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
      * When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
      * The filter text should match any part of the item name (i.e. it should _not_ only match from the start of the string)
      Notes:
      * The wireframe shows the filter field, but not it being used. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png
      * The following example app from MDN is a great example of providing feedback that is accessible to screenreaders when things change on the screen. https://moz-todomvc-react.netlify.app/

### PR DESCRIPTION
**Pull request by vividavila98**
## Description

In this PR, we added the ability to filter your shopping list based on what is typed in the input field. 

To do this, we created the state `filterText` that holds the value of the input field. We then filtered the collection array (held in the `value` state) to only include the documents whose `item` field in the db includes the same characters that makeup `filterText`. We made this filtering functionality case-insensitive by lowercasing `filterText` and `doc.data().item` when making the comparison. With this newly filtered collection array, we mapped over each document and displayed them in an `li` tag. 

Additionally, we added a button next to the input field to allow the user to clear out the text. 

## Related Issue

closes #9 

## Acceptance Criteria
- [X] Display a text field above the top of the shopping list
- [X] As the user types into the field, the list should narrow to display only items that contain the text the user entered in the filter field
- [X] When the field has text in it, the user should be able to tap a UI element (e.g., with an "X" button next to the field) to clear the field
- [X] The filter text should match any part of the item name (i.e. it should not only match from the start of the string)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Video: Walk-Through

https://user-images.githubusercontent.com/76231436/152875800-999dda41-2222-4a92-9d2f-02e684bd33de.mp4

## Testing Steps / QA Criteria

1. `npm start`
2. Make sure your collection has items with the same letters in them: Pie, apple, chorizo
3. If you type the letter "p", you should expect the list: Pie, apple 
4. If you click the X button, you should expect the input field to clear.

### CODE REVIEW

**Review by danainjax — APPROVED** [review]
This works great! 🎉

🎬  I also really like how you refactored the code on lines 73, 74, and 30, to make it cleaner with the anonymous function. 

Suggestion: You may want to render the list filter conditionally. Right now I see the filter whether there are items in the list or not.

Awesome job completing the sprint on the first day! 🏃🏾‍♀️👏



Screen Shot 2022-02-09 at 3 23 51 PM

**Review by sanrafa — APPROVED** [review]
Great work, y'all! I agree with Dana's suggestion that it might be a good idea to render the search field conditionally, it's approved :)

File: src/components/List.jsx [hunk 1]
```diff
@@ -27,32 +28,54 @@ export const List = ({ token }) => {
...
+
+      <div style={{ marginTop: '1em' }}>
+        <input
+          placeholder="Start typing here..."
+          value={filterText}
+          onChange={handleFilterChange}
+        />
+        <button style={{ marginLeft: '10px' }} onClick={() => clearFilter()}>
+          X
+        </button>
```
Comment (yenly) [inline src/components/List.jsx]: Not a must change. For future reference, you can call the setFilterText function in the onClick
```suggestion
        <button style={{ marginLeft: '10px' }} onClick={() => setFilterText('')}>
          X
        </button>
```

**Review by yenly — APPROVED** [review]
👏🏽 Great work! I like @danainjax and @sanrafa's suggestion.

**Pull request merged**